File: 백준/Silver/1475. 방 번호/방 번호.java


import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {

    private static int n;
    private static int[] numCnt = new int[10];
    private static final int ZERO = 48;

    public static void main(String[] args) throws IOException {
        
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String input = br.readLine();

        for(int i = 0; i < input.length(); i++) {

            int cur = input.charAt(i) - ZERO;
            numCnt[cur]++;
        }

        int sixPlusNine = 0;
        int maxValue = 0;
        for(int i = 0; i < numCnt.length; i++) {

            if(i == 6 || i == 9) {
                sixPlusNine+=numCnt[i];
                continue;
            }
            maxValue = Math.max(numCnt[i], maxValue);
        }

        if(sixPlusNine % 2 > 0){
            sixPlusNine = (sixPlusNine / 2)+1;
        } else {
            sixPlusNine = sixPlusNine / 2;
        }

        maxValue = Math.max(maxValue, sixPlusNine);

        System.out.println(maxValue);
    }
}

File: 백준/Silver/1475. 방 번호/방 번호.py

room = int(input())

nums = list(str(room))
arr = [0] * 10

for num in nums:
    arr[int(num)] += 1

six_plus_nine = 0
max_value = 0
for i in range(len(arr)):
    if i == 6 or i == 9:
        six_plus_nine += arr[i]
    else:
        max_value = max(arr[i], max_value)

six_plus_nine = (six_plus_nine // 2) + (six_plus_nine % 2)
max_value = max(max_value, six_plus_nine)

print(max_value)